.github/workflows/release.yaml

```diff
@@ -0,0 +1,230 @@
+          go install google.golang.org/protobuf/cmd/protoc-gen-go@32051b4f86e54c2142c7c05362c6e96ae3454a1c # @v1.28.0
+          go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@938f6e2f7550e542bd78f3b9e8812665db109e02 # @v1.1.0
+          make bin/protoc bin/goreleaser
+          bash ./dev/run-kind-registry.sh
+          make ebpf generate
+          ./bin/goreleaser release --snapshot --rm-dist
+          make push-local-images
+          cp dist/tarianctl_linux_amd64/tarianctl ./bin/
+
+      - name: Run unit tests
+        run: make unit-test
+
+      - name: Install Tarian
+        run: |
+          set -x
+          ./bin/tarianctl install --charts ./charts -l debug --agents-values ./dev/values/agents.yaml --server-values ./dev/values/server.yaml
+        continue-on-error: true
+
+      - name: Wait for pods to be ready
+        id: wait-until-tarian-is-ready
+        run: |
+          set -x
+          # Wait for all pods in the tarian-system namespace to be in the "Ready" state
+          kubectl wait --for=condition=ready pod --all -n tarian-system --timeout=300s
+        continue-on-error: true
+
+      - name: Run tests
+        id: run-tests
+        if: steps.wait-until-tarian-is-ready.outcome == 'success'
+        run: make k8s-test
+        continue-on-error: true
+
+      - name: Collect deployment information in case of failure
+        if: steps.wait-until-tarian-is-ready.outcome != 'success' || steps.run-tests.outcome != 'success'
+        run: |
+          set -x
+          # Collect logs and information for debugging in case of failure
+          kubectl get pods -n tarian-system
+          echo "======================================================================================================================"
+          kubectl describe pods -n tarian-system
+          echo "======================================================================================================================"
+          kubectl logs -n tarian-system deploy/tarian-server
+          echo "======================================================================================================================"
+          kubectl logs -n tarian-system deploy/tarian-cluster-agent
+          echo "======================================================================================================================"
+          kubectl logs -n tarian-system deploy/tarian-controller-manager
+          echo "======================================================================================================================"
+          kubectl logs -n tarian-system daemonsets/tarian-node-agent
+          echo "======================================================================================================================"
+          exit 1
+
+  publish-containers:
+    needs: [unit-test, test-k8s]
+    if: ${{ github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v') }}
```

Comment:
You raised an idea that we don't need to publish containers for new commits to main, only publish on release. However, this code runs on push to main. What do you think now?